packages/query/src/block-processor.ts

```diff
@@ -142,10 +148,6 @@ export class BlockProcessor implements BlockProcessorInterface {
       keepAlive: true,
       abortSignal: this.abortController.signal,
     })) {
-      if (nextEpochStartHeight === compactBlock.height) {
-        await this.indexedDb.addEpoch(nextEpochStartHeight);
-        nextEpochStartHeight = undefined;
-      }
```

Comment:
This is now moved to the end of this loop, to avoid the edge case @Valentine1898 discovered [here](https://github.com/penumbra-zone/web/pull/699#discussion_r1515328606).